### PR TITLE
Allow night mode to follow system in Android M and higher

### PR DESCRIPTION
**Pull request by PiwwowPants**
I made night mode follow the system on Android M and higher.
I additionally hide the options to set your own time if you're on M or higher so the system can just do this for us.

If you'd like the options to retain, I can keep them. But this is without. With, I'd have to add another option to this menu, which is not a big deal but I hate adding options.

Let me know either way and I can make the change if desired.

### CODE REVIEW

File: app/src/main/java/me/ccrama/redditslide/SettingValues.java [hunk 1]
```diff
@@ -468,8 +470,17 @@ public static TimePeriod getSubmissionTimePeriod(String sub) {
 
 
     public static boolean isNight() {
-        int hour = Calendar.getInstance().get(Calendar.HOUR_OF_DAY);
-        return (hour >= nightStart + 12 || hour < nightEnd) && tabletUI && nightMode;
```
Comment (PiwwowPants) [inline app/src/main/java/me/ccrama/redditslide/SettingValues.java]: I removed the check for `tabletUI`. It felt wrong, but if it is not, let me know and I can put it back.

**Comment by PiwwowPants** [comment]
I think the build erroneously canceled

**Comment by PiwwowPants** [comment]
Tested through an entire day. Works great.

**Comment by PiwwowPants** [comment]
Just pushed a bad commit message description. Same change.

**Comment by ccrama** [comment]
Same as #2781 :D

**Comment by ccrama** [comment]
Beautiful, thank you! Now just waiting on Google to put in a system night mode this will pair nicely

**Pull request merged**